tests/test_azure_storage_api.py

```diff
@@ -37,41 +46,43 @@ def test_mount_existing_container(self, MockFromConnectionString):
         self.assertEqual(result, mock_container_client)
 
     @patch("azure.storage.blob.BlobServiceClient.from_connection_string")
-    def test_mount_nonexisting_container_create(self, MockFromConnectionString):
-        """
-        tests when a container does not exists and create_container flag is set to True,
-        should create a new container and return the container client
-        """
-        # mock the client container and blob service client
+    @patch("azure_storage.azure_storage_api.get_directories")
+    def test_mount_nonexisting_container_create(self, mock_get_directories, MockFromConnectionString):
+        # Mock the BlobServiceClient
+        mock_blob_service_client = MockFromConnectionString.return_value
+
+        # Mock the container client
         mock_container_client = Mock()
         mock_container_client.exists.return_value = False
 
-        mock_blob_service_client = MockFromConnectionString.return_value
-        mock_blob_service_client.get_container_client.return_value = (
-            mock_container_client
-        )
+        # Set the return value for get_container_client
+        mock_blob_service_client.get_container_client.return_value = mock_container_client
 
         # Simulate that a new container is created
         mock_new_container_client = Mock()
-        mock_blob_service_client.create_container.return_value = (
-            mock_new_container_client
-        )
+        mock_blob_service_client.create_container.return_value = mock_new_container_client
+
+        # Mock get_directories to return an empty dictionary
+        mock_get_directories.return_value = {}
 
+        # Test setup
         connection_string = "test_connection_string"
         container_name = "testcontainer"
         expected_container_name = "user-{}".format(container_name)
 
+        # Run the test
         loop = asyncio.new_event_loop()
         asyncio.set_event_loop(loop)
-        result = loop.run_until_complete(
-            mount_container(connection_string, container_name, create_container=True)
-        )
+        try:
+            result = loop.run_until_complete(
```

Comment:
why are you not using self.loop here?

Reply:
I mistakenly didn't adapt the tests from the first class.